lib/rules/no-plusplus.js

```diff
@@ -6,6 +6,34 @@
 
 "use strict";
 
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Determines whether the given node is the update node of a `ForStatement`.
+ * @param {ASTNode} node The node to check.
+ * @returns {boolean} `true` if the node is `ForStatement` update.
+ */
+function isForStatementUpdate(node) {
+    const parent = node.parent;
+
+    return parent.type === "ForStatement" && parent.update === node;
+}
+
+/**
+ * Determines whether the given node is considered to be a for loop "afterthought" by the logic of this rule.
+ * In particular, if it is the update node of a `ForStatement`, or an operand of a sequence expression that is the update node.
+ * @param {ASTNode} node The node to check.
+ * @returns {boolean} `true` if the node is a for loop afterthought.
+ */
+function isForLoopAfterthought(node) {
```

Comment:
As @yeonjuan mentioned, it appears that this check only looks up one level, which could mean some patterns will still be flagged as incorrect.

```js
for (;; (++a, (++b, ++c) ) {}  // it's warned in the PR version. no-plusplus for `b` and `c`
```

It may be an unlikely case, but for the sake of completeness, it probably makes sense to continue looking up the AST if the parent is a `SequenceExpression`.

Reply:
It's done now! It makes sense to me, too.